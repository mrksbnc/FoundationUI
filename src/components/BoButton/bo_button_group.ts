export enum BoButtonGroupSize {
	small = 'small',
	default = 'default',
	large = 'large',
}

export enum BoButtonGroupVariant {
	default = 'default',
	dark = 'dark',
	info = 'info',
	success = 'success',
	warning = 'warning',
	danger = 'danger',
}

export enum ButtonGroupEvent {
	/**
	 * @event select
	 * @description Emitted when one of the buttons in the group is clicked
	 * @emits `{ id: string | number, item: BoButtonGroupItem }` The id and the item of the selected button
	 */
	SELECT = 'select',
}
