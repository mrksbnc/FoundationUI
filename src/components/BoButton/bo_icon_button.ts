export enum BoIconButtonType {
	default = 'default',
	round = 'round',
}
