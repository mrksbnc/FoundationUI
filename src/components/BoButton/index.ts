/**
 * @module BoButton
 * @since 0.0.1
 *
 * @description module export for the BoButton component and its members
 */
export { BoButtonType, BoButtonVariant, useBoButton } from './ts/bo_button';
export type {
	BoButtonClassMap,
	UseBoButtonCompoableArgs,
} from './ts/bo_button.types';
export { default as BoButton } from './vue/bo_button.vue';
