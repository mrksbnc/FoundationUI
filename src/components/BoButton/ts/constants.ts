export enum BoButtonVariant {
	primary = 'primary',
	secondary = 'secondary',
	danger = 'danger',
	warning = 'warning',
	success = 'success',
	dark = 'dark',
	purple = 'purple',
	teal = 'teal',
}

export enum BoButtonShadow {
	blue = 'blue',
	gray = 'gray',
	red = 'red',
	yellow = 'yellow',
	green = 'green',
	black = 'black',
	purple = 'purple',
	teal = 'teal',
}

export enum BoButtonType {
	default = 'default',
	pill = 'pill',
	outline = 'outline',
}

export enum BoIconButtonType {
	default = 'default',
	round = 'round',
}

export enum BoButtonGroupSize {
	small = 'small',
	default = 'default',
	large = 'large',
}

export enum BoButtonGroupVariant {
	default = 'default',
	dark = 'dark',
	info = 'info',
	success = 'success',
	warning = 'warning',
	danger = 'danger',
}
