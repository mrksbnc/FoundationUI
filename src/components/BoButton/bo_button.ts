import {
	type BoButtonClassMap,
	type UseBoButtonCompoableArgs,
} from '@/components/BoButton';
import { BoSize } from '@/constants';
import { TailwindUtils } from '@/utils';
import { computed } from 'vue';

export enum BoButtonType {
	default = 'default',
	pill = 'pill',
	outline = 'outline',
}

export enum BoButtonVariant {
	primary = 'primary',
	secondary = 'secondary',
	success = 'success',
	warning = 'warning',
	danger = 'danger',
	dark = 'dark',
	purple = 'purple',
	teal = 'teal',
}

const BO_BUTTON_TRANSITION_CLASSES =
	/*tw*/ 'transition-all duration-200 ease-in-out' as const;

const BO_BUTTON_FLEX_CLASSES =
	/*tw*/ 'flex items-center justify-center content-center text-center w-full max-w-fit space-x-1' as const;

const BO_BUTTON_DISABLED_CLASSES =
	/*tw*/ 'disabled:cursor-not-allowed disabled:opacity-50' as const;

const BO_BUTTON_DEFAULT_CLASSES: BoButtonClassMap<BoButtonVariant> =
	Object.freeze({
		default: {
			primary: /*tw*/ 'text-white dark:text-white bg-blue-700 dark:bg-blue-600',
			secondary:
				/*tw*/ 'text-gray-600 dark:text-gray-500 bg-gray-200 dark:bg-gray-100',
			success:
				/*tw*/ 'text-white dark:text-white bg-green-600 dark:bg-green-500',
			warning:
				/*tw*/ 'text-white dark:text-white bg-yellow-600 dark:bg-yellow-500',
			danger: /*tw*/ 'text-white dark:text-white bg-red-600 dark:bg-red-500',
			dark: /*tw*/ 'text-white dark:text-white bg-gray-600 dark:bg-gray-500',
			purple:
				/*tw*/ 'text-white dark:text-white bg-purple-600 dark:bg-purple-500',
			teal: /*tw*/ 'text-white dark:text-white bg-teal-600 dark:bg-teal-500',
		},
		active: {
			primary:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-blue-800 dark:hover:bg-blue-700',
			secondary:
				/*tw*/ 'hover:text-gray-800 dark:hover:text-gray-300 hover:bg-gray-300 dark:hover:bg-gray-200',
			success:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-green-700 dark:hover:bg-green-600',
			warning:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-yellow-700 dark:hover:bg-yellow-600',
			danger:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-red-700 dark:hover:bg-red-600',
			dark: /*tw*/ 'hover:text-gray-600 dark:hover:text-white hover:bg-gray-700 dark:hover:bg-gray-600',
			purple:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-purple-700 dark:hover:bg-purple-600',
			teal: /*tw*/ 'hover:text-white dark:hover:text-white hover:bg-teal-700 dark:hover:bg-teal-600',
		},
	});

const BO_BUTTON_OUTLINE_CLASSES: BoButtonClassMap<BoButtonVariant> =
	Object.freeze({
		default: {
			primary:
				/*tw*/ 'border border-blue-700 dark:border-blue-600 text-blue-700 dark:text-blue-600 bg-transparent',
			secondary:
				/*tw*/ 'border border-gray-700 dark:border-gray-600 text-gray-700 dark:text-gray-600 bg-transparent',
			success:
				/*tw*/ 'border border-green-600 dark:border-green-500 text-green-600 dark:text-green-500 bg-transparent',
			warning:
				/*tw*/ 'border border-yellow-600 dark:border-yellow-500 text-yellow-600 dark:text-yellow-500 bg-transparent',
			danger:
				/*tw*/ 'border border-red-600 dark:border-red-500 text-red-600 dark:text-red-500 bg-transparent',
			dark: /*tw*/ 'border border-gray-600 dark:border-gray-500 text-gray-600 dark:text-gray-500 bg-transparent',
			purple:
				/*tw*/ 'border border-purple-600 dark:border-purple-500 text-purple-600 dark:text-purple-500 bg-transparent',
			teal: /*tw*/ 'border border-teal-600 dark:border-teal-500 text-teal-600 dark:text-teal-500 bg-transparent',
		},
		active: {
			primary:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-blue-700 dark:hover:bg-blue-600',
			secondary:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-gray-500 dark:hover:bg-gray-500 hover:border-gray-500 dark:hover:border-gray-500',
			success:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-green-600 dark:hover:bg-green-500',
			warning:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-yellow-600 dark:hover:bg-yellow-500',
			danger:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-red-600 dark:hover:bg-red-500',
			dark: /*tw*/ 'hover:text-white dark:hover:text-white hover:bg-gray-600 dark:hover:bg-gray-500',
			purple:
				/*tw*/ 'hover:text-white dark:hover:text-white hover:bg-purple-600 dark:hover:bg-purple-500',
			teal: /*tw*/ 'hover:text-white dark:hover:text-white hover:bg-teal-600 dark:hover:bg-teal-500',
		},
	});

enum BoButtonBorderRadiusClasses {
	pill = /*tw*/ 'rounded-full',
	default = /*tw*/ 'rounded-lg',
	outline = /*tw*/ 'rounded-lg',
}

enum BoButtonPaddingClasses {
	extra_small = /*tw*/ 'px-2 py-1',
	small = /*tw*/ 'px-2 py-1.5',
	default = /*tw*/ 'px-3 py-2',
	large = /*tw*/ 'px-4 py-3',
	extra_large = /*tw*/ 'px-5 py-4',
}

enum BoButtonTextSizeClasses {
	extra_small = /*tw*/ 'text-xs',
	small = /*tw*/ 'text-xs',
	default = /*tw*/ 'text-sm',
	large = /*tw*/ 'text-base',
	extra_large = /*tw*/ 'text-lg',
}

enum BoButtonShadowClasses {
	blue = /*tw*/ 'shadow-sm shadow-blue-500/50 dark:shadow-sm dark:shadow-blue-800/80',
	gray = /*tw*/ 'shadow-sm shadow-gray-500/50 dark:shadow-sm dark:shadow-gray-800/80',
	red = /*tw*/ 'shadow-sm shadow-red-500/50 dark:shadow-sm dark:shadow-red-800/80',
	yellow = /*tw*/ 'shadow-sm shadow-yellow-500/50 dark:shadow-sm dark:shadow-yellow-800/80',
	green = /*tw*/ 'shadow-sm shadow-green-500/50 dark:shadow-sm dark:shadow-green-800/80',
	black = /*tw*/ 'shadow-sm shadow-black-500/50 dark:shadow-sm dark:shadow-black-800/80',
	purple = /*tw*/ 'shadow-sm shadow-purple-500/50 dark:shadow-sm dark:shadow-purple-800/80',
	teal = /*tw*/ 'shadow-sm shadow-teal-500/50 dark:shadow-sm dark:shadow-teal-800/80',
}

export const useBoButtonStyle = (
	props: UseBoButtonCompoableArgs,
): { classes: string; iconSize: BoSize } => {
	const variant = props.variant.value;
	const type = props.type.value;
	const size = props.size.value;

	const isOutline = computed<boolean>(() => type === BoButtonType.outline);

	const base = computed<string>(() => {
		let shadow = '';

		switch (variant) {
			case BoButtonVariant.primary:
				shadow = BoButtonShadowClasses.blue;
				break;
			case BoButtonVariant.secondary:
				shadow = BoButtonShadowClasses.gray;
				break;
			case BoButtonVariant.success:
				shadow = BoButtonShadowClasses.green;
				break;
			case BoButtonVariant.warning:
				shadow = BoButtonShadowClasses.yellow;
				break;
			case BoButtonVariant.danger:
				shadow = BoButtonShadowClasses.red;
				break;
			case BoButtonVariant.dark:
				shadow = BoButtonShadowClasses.black;
				break;
			case BoButtonVariant.purple:
				shadow = BoButtonShadowClasses.purple;
				break;
			case BoButtonVariant.teal:
				shadow = BoButtonShadowClasses.teal;
				break;
		}

		return isOutline.value
			? TailwindUtils.merge(
					BO_BUTTON_FLEX_CLASSES,
					BO_BUTTON_DISABLED_CLASSES,
					BO_BUTTON_TRANSITION_CLASSES,
					BO_BUTTON_OUTLINE_CLASSES.default[variant],
					BO_BUTTON_OUTLINE_CLASSES.active != null
						? BO_BUTTON_OUTLINE_CLASSES.active[variant]
						: '',
				)
			: TailwindUtils.merge(
					shadow,
					BO_BUTTON_FLEX_CLASSES,
					BO_BUTTON_DISABLED_CLASSES,
					BO_BUTTON_TRANSITION_CLASSES,
					BO_BUTTON_DEFAULT_CLASSES.default[variant],
					BO_BUTTON_DEFAULT_CLASSES.active != null
						? BO_BUTTON_DEFAULT_CLASSES.active[variant]
						: '',
				);
	});

	const borderRadius = computed<string>(() => {
		return BoButtonBorderRadiusClasses[type];
	});

	const buttonAndContentSize = computed<string>(() => {
		switch (size) {
			case BoSize.extra_small:
				return TailwindUtils.merge(
					BoButtonPaddingClasses.extra_small,
					BoButtonTextSizeClasses.extra_small,
				);
			case BoSize.small:
				return TailwindUtils.merge(
					BoButtonPaddingClasses.small,
					BoButtonTextSizeClasses.small,
				);
			case BoSize.large:
				return TailwindUtils.merge(
					BoButtonPaddingClasses.large,
					BoButtonTextSizeClasses.large,
				);
			case BoSize.extra_large:
				return TailwindUtils.merge(
					BoButtonPaddingClasses.extra_large,
					BoButtonTextSizeClasses.extra_large,
				);
			case BoSize.default:
			default:
				return TailwindUtils.merge(
					BoButtonPaddingClasses.default,
					BoButtonTextSizeClasses.default,
				);
		}
	});

	const classes = computed<string>(() => {
		return TailwindUtils.merge(
			base.value,
			borderRadius.value,
			buttonAndContentSize.value,
		);
	});

	const iconSize = computed<BoSize>(() => {
		switch (size) {
			case BoSize.large:
			case BoSize.extra_large:
				return BoSize.default;
			case BoSize.extra_small:
			case BoSize.small:
			case BoSize.default:
			default:
				return BoSize.small;
		}
	});

	return {
		classes: classes.value,
		iconSize: iconSize.value,
	};
};
