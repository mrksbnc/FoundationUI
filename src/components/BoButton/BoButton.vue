<template>
	<button
		type="button"
		:disabled="disabled"
		:class="[`bo-button--${variant}`, style]"
	>
		<span class="bo-button__prefix">
			<BoIcon
				v-if="prefixIcon"
				:icon="prefixIcon"
			/>
		</span>
		<span class="bo-button__label">
			{{ label }}
		</span>
		<span class="bo-button__suffix">
			<BoIcon
				v-if="suffixIcon"
				:icon="suffixIcon"
			/>
		</span>
	</button>
</template>

<script setup lang="ts">
import { computed, toRefs, type PropType } from 'vue';
import { BoButtonType, BoButtonVariant } from './bo_button';
import { BoSize } from '@/enums/size';
import BoIcon from '../BoIcon/BoIcon.vue';

const props = defineProps({
	label: {
		type: String,
		required: true,
	},
	variant: {
		type: String as PropType<BoButtonVariant>,
		default: BoButtonVariant.primary,
	},
	type: {
		type: String as PropType<BoButtonType>,
		default: BoButtonType.default,
	},
	disabled: {
		type: Boolean,
		default: false,
	},
	size: {
		type: String as PropType<BoSize>,
		default: BoSize.default,
	},
	prefixIcon: {
		type: String,
		default: null,
	},
	suffixIcon: {
		type: String,
		default: null,
	},
});

const { label, variant, type, disabled, size, prefixIcon, suffixIcon } =
	toRefs(props);

const disabledClasses = 'disabled:cursor-not-allowed disabled:opacity-50';

const sizeClasses = computed(() => {
	switch (size.value) {
		case BoSize.extra_small:
			return 'px-3 py-2';
		case BoSize.small:
			return 'px-3 py-2';
		case BoSize.large:
			return 'px-5 py-3';
		case BoSize.extra_large:
			return 'px-6 py-3.5';
		case BoSize.default:
		default:
			return 'px-5 py-2.5';
	}
});

const colorClasses = computed(() => {
	switch (type.value) {
		case BoButtonType.outline:
			switch (variant.value) {
				case BoButtonVariant.danger:
					return 'border border-red-500 text-white hover:bg-red-600 focus:ring-red-500';
				case BoButtonVariant.warning:
					return 'border-yellow-500 text-white hover:bg-yellow-600 focus:ring-yellow-500';
				case BoButtonVariant.success:
					return 'bg-green-500 text-white hover:bg-green-600 focus:ring-green-500';
				case BoButtonVariant.dark:
					return 'bg-gray-800 text-white hover:bg-gray-900 focus:ring-gray-800';
				case BoButtonVariant.light:
					return 'bg-gray-200 text-gray-800 hover:bg-gray-300 focus:ring-gray-200';
				case BoButtonVariant.link:
					return 'text-blue-500 hover:text-blue-600 focus:ring-blue-500';
				case BoButtonVariant.alternative:
					return 'bg-gray-200 text-gray-800 hover:bg-gray-300 focus:ring-gray-200';
				case BoButtonVariant.secondary:
					return 'bg-gray-500 text-white hover:bg-gray-600 focus:ring-gray-500';
				case BoButtonVariant.primary:
				default:
					return 'bg-blue-500 text-white hover:bg-blue-600 focus:ring-blue-500';
			}
		case BoButtonType.pill:
		case BoButtonType.default:
		default:
			switch (variant.value) {
				case BoButtonVariant.danger:
					return 'bg-red-500 text-white hover:bg-red-600 focus:ring-red-500';
				case BoButtonVariant.warning:
					return 'bg-yellow-500 text-white hover:bg-yellow-600 focus:ring-yellow-500';
				case BoButtonVariant.success:
					return 'bg-green-500 text-white hover:bg-green-600 focus:ring-green-500';
				case BoButtonVariant.dark:
					return 'bg-gray-800 text-white hover:bg-gray-900 focus:ring-gray-800';
				case BoButtonVariant.light:
					return 'bg-gray-200 text-gray-800 hover:bg-gray-300 focus:ring-gray-200';
				case BoButtonVariant.link:
					return 'text-blue-500 hover:text-blue-600 focus:ring-blue-500';
				case BoButtonVariant.alternative:
					return 'bg-gray-200 text-gray-800 hover:bg-gray-300 focus:ring-gray-200';
				case BoButtonVariant.secondary:
					return 'bg-gray-500 text-white hover:bg-gray-600 focus:ring-gray-500';
				case BoButtonVariant.primary:
				default:
					return 'bg-blue-500 text-white hover:bg-blue-600 focus:ring-blue-500';
			}
	}
});

const style = computed(() => {
	return `${sizeClasses.value} ${colorClasses.value} ${disabled.value ? disabledClasses : ''}`;
});
</script>
