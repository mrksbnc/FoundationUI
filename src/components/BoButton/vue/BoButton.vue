<template>
	<button
		type="button"
		role="button"
		:disabled="disabled"
		:class="[classes, `bo-button--${variant} bo-button--${type}`]"
	>
		<span
			class="bo-button__prefix"
			v-if="prefixIcon || $slots.prefix"
		>
			<bo-icon
				v-if="!$slots.prefix"
				:icon="prefixIcon"
				:size="iconSize"
			/>
			<slot name="prefix"></slot>
		</span>
		<span class="bo-button__label mx-1 font-medium shadow-none">
			{{ label }}
		</span>
		<span
			class="bo-button__suffix"
			v-if="suffixIcon || $slots.suffix"
		>
			<bo-icon
				v-if="!$slots.suffix"
				:icon="suffixIcon"
				:size="iconSize"
			/>
			<slot name="suffix"></slot>
		</span>
	</button>
</template>

<script setup lang="ts">
import { computed, toRefs, type PropType } from 'vue';
import { BoButtonType } from '@/components/BoButton';
import { BoIcon, type Icon } from '@/components/BoIcon';
import { BoSize } from '@/constants';

import { useBoButtonStyle, BoButtonVariant } from '@/components/BoButton';

const props = defineProps({
	label: {
		type: String,
		required: true,
	},
	variant: {
		type: String as PropType<BoButtonVariant>,
		default: BoButtonVariant.primary,
	},
	type: {
		type: String as PropType<BoButtonType>,
		default: BoButtonType.default,
	},
	disabled: {
		type: Boolean,
		default: false,
	},
	size: {
		type: String as PropType<BoSize>,
		default: BoSize.default,
	},
	prefixIcon: {
		type: String as PropType<Icon>,
		default: null,
	},
	suffixIcon: {
		type: String as PropType<Icon>,
		default: null,
	},
});

const classes = computed(() => useBoButtonStyle(toRefs(props)).classes);
const iconSize = computed(() => useBoButtonStyle(toRefs(props)).iconSize);
</script>
