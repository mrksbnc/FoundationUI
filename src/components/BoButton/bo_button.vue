<template>
	<button
		type="button"
		role="button"
		:disabled="disabled"
		:class="[classes, `bo-button--${variant} bo-button--${type}`]"
	>
		<span
			class="bo-button__prefix"
			v-if="prefixIcon || $slots.prefix"
		>
			<div v-if="!$slots.prefix">
				<bo-icon
					role="img"
					:size="iconSize"
					:icon="prefixIcon"
				/>
			</div>
			<div v-else>
				<slot name="prefix"></slot>
			</div>
		</span>
		<span
			role="text"
			class="bo-button__label font-medium shadow-none"
		>
			{{ label }}
		</span>
		<span
			class="bo-button__suffix"
			v-if="suffixIcon || $slots.suffix"
		>
			<div v-if="!$slots.suffix">
				<bo-icon
					role="img"
					:size="iconSize"
					:icon="suffixIcon"
				/>
			</div>
			<div v-else>
				<slot name="suffix"></slot>
			</div>
		</span>
	</button>
</template>

<script setup lang="ts">
import { computed, toRefs, type PropType } from 'vue';
import { BoIcon, type Icon } from '@/components/BoIcon';
import { BoSize } from '@/constants';
import {
	useBoButton,
	BoButtonVariant,
	BoButtonType,
} from '@/components/BoButton';

const props = defineProps({
	label: {
		type: String,
		required: true,
	},
	variant: {
		type: String as PropType<BoButtonVariant>,
		default: BoButtonVariant.primary,
	},
	type: {
		type: String as PropType<BoButtonType>,
		default: BoButtonType.default,
	},
	disabled: {
		type: Boolean,
		default: false,
	},
	size: {
		type: String as PropType<BoSize>,
		default: BoSize.default,
	},
	prefixIcon: {
		type: String as PropType<Icon>,
		default: null,
	},
	suffixIcon: {
		type: String as PropType<Icon>,
		default: null,
	},
});

const classes = computed(() => useBoButton(toRefs(props)).classes);
const iconSize = computed(() => useBoButton(toRefs(props)).iconSize);
</script>
