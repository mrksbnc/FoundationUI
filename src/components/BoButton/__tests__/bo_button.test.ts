import { BoButton, BoButtonType } from '@/components/BoButton';
import { BoSize, BoVariant } from '@/constants';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let wrapper: ReturnType<typeof mount>;

beforeEach(() => {
	wrapper = mount(BoButton, {
		props: {
			label: 'Label',
			size: BoSize.default,
			disabled: false,
			type: BoButtonType.default,
			variant: BoVariant.primary,
		},
	});
});

describe('BoButton', () => {
	test('should match snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('should render the label', () => {
		expect(wrapper.text()).toBe('Label');
	});

	test('should have the default size', () => {
		expect(wrapper.find('px-3')).toBeTruthy();
		expect(wrapper.find('py-2')).toBeTruthy();
	});

	test('should have the default type', () => {
		expect(wrapper.find('bg-blue-500')).toBeTruthy();
		expect(wrapper.find('rounded-lg')).toBeTruthy();
	});

	test('should have the default text classes', () => {
		expect(wrapper.find('text-white')).toBeTruthy();
		expect(wrapper.find('text-sm')).toBeTruthy();
	});

	test('should be able to change the size', async () => {
		await wrapper.setProps({ size: BoSize.large });
		expect(wrapper.find('px-4')).toBeTruthy();
		expect(wrapper.find('py-3')).toBeTruthy();
	});

	test('should be able to change the type', async () => {
		await wrapper.setProps({ type: BoButtonType.outline });
		expect(wrapper.find('border')).toBeTruthy();
	});
});
