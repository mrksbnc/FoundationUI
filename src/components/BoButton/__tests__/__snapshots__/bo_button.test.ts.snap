// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BoButton > should match snapshot 1`] = `
"<button type="button" class="rounded-lg px-3 py-2 bg-blue-500 hover:bg-blue-600 focus:ring-blue-500 disabled:cursor-not-allowed disabled:opacity-50 text-white text-sm bo-button--primary shadow-md focus:ring-1 focus:ring-opacity-50 focus:outline-none transition-all duration-200 ease-in-out flex items-center justify-center space-x-2">
  <!--v-if--><span class="bo-button__label">Label</span>
  <!--v-if-->
</button>"
`;
